src/finch/interface/lazy.py

```diff
@@ -393,3 +393,79 @@ def positive(x) -> LazyTensor:
 
 def negative(x) -> LazyTensor:
     return elementwise(operator.neg, defer(x))
+
+
+def any(
+    x,
+    /,
+    *,
+    axis: int | tuple[int, ...] | None = None,
+    keepdims: bool = False,
+    init=None,
+):
+    """
+    Test whether any element of input array ``x`` along given axis is True.
+    """
+    if init is None:
+        init = init_value(builtins.any, x)
+    return reduce(
+        operator.or_,
+        elementwise(operator.truth, x),
+        axis=axis,
+        keepdims=keepdims,
+        init=init,
+    )
+
+
+def all(
+    x,
+    /,
+    *,
+    axis: int | tuple[int, ...] | None = None,
+    keepdims: bool = False,
+    init=None,
+):
+    """
+    Test whether all elements of input array ``x`` along given axis are True.
+    """
+    if init is None:
+        init = init_value(builtins.all, x)
```

Comment:
I think you mean to use `operator.and_`

Reply:
This is not resolved, the reduce function already defines an init. just pass init unmodified.